remote-api/src/main/java/org/alfresco/rest/api/impl/rules/ActionPermissionValidator.java

```
@@ -0,0 +1,85 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.rest.api.impl.rules;
+
+import static org.alfresco.repo.web.scripts.rule.AbstractRuleWebScript.CANNOT_CREATE_RULE;
+import static org.alfresco.service.cmr.rule.RuleType.OUTBOUND;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.alfresco.repo.action.CompositeActionImpl;
+import org.alfresco.repo.action.RuntimeActionService;
+import org.alfresco.repo.action.access.ActionAccessRestriction;
+import org.alfresco.repo.action.executer.CheckOutActionExecuter;
+import org.alfresco.rest.framework.core.exceptions.InvalidArgumentException;
+import org.alfresco.service.Experimental;
+import org.alfresco.service.cmr.action.Action;
+import org.alfresco.service.cmr.rule.Rule;
+import org.apache.commons.collections.CollectionUtils;
+
+@Experimental
+public class ActionPermissionValidator
+{
+    private final RuntimeActionService runtimeActionService;
+
+    public ActionPermissionValidator(RuntimeActionService runtimeActionService)
+    {
+        this.runtimeActionService = runtimeActionService;
+    }
+
+    Rule validateRulePermissions(Rule rule)
+    {
+        final List<Action> actions = ((CompositeActionImpl) rule.getAction()).getActions();
+
+        checkRestrictedAccessActions(actions);
+        checkRuleOutboundHasNoCheckOutAction(rule, actions);
+        return rule;
+    }
+
+    private void checkRestrictedAccessActions(List<Action> actions) {
+        actions.forEach(action -> {
+            ActionAccessRestriction.setActionContext(action, ActionAccessRestriction.RULE_ACTION_CONTEXT);
+            runtimeActionService.verifyActionAccessRestrictions(action);
+        });
+    }
+
+    private void checkRuleOutboundHasNoCheckOutAction(Rule rule, List<Action> actions) {
+        //TODO: rule types should never be empty in final implementation
```

I'm not sure what needs to be "TODONE" here. Are you planning to create a second PR to address it, or is there another Jira ticket maybe?

---

I guess we should throw an exception when `rule.ruleTypes` is null or empty.